<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.36">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Slamka taxdist project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Slamka taxdist project</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="prepare-data" class="level2">
<h2 class="anchored" data-anchor-id="prepare-data">Prepare data</h2>
<div class="cell">

</div>
<p>We examined 113 samples, collected from a healthy volunteer (<code>HV</code>), and 3 cancer patients undergoing cancer treatment (<code>PT1</code>, <code>PT2</code>, <code>PT3</code>). In each subject, multiple samples were collected.</p>
<div class="cell">
<div class="cell-output-display">
<table class="table table-sm table-striped">
<caption>Subjects and number of samples </caption>
<thead>
<tr class="header">
<th style="text-align: left;">pt</th>
<th style="text-align: right;">n.samples</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">HV</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="even">
<td style="text-align: left;">PT1</td>
<td style="text-align: right;">40</td>
</tr>
<tr class="odd">
<td style="text-align: left;">PT2</td>
<td style="text-align: right;">41</td>
</tr>
<tr class="even">
<td style="text-align: left;">PT3</td>
<td style="text-align: right;">21</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Each subject’s samples were collected on different days, but also collected on the same day. Some samples were also sequenced multiple times.</p>
<div class="cell" data-fig.asp="0.3">
<div class="cell-output-display">
<p><img src="README_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
<p>By selecting all possible pairwise combinations of the sequencing data, we generated various types of two-sample comparisons:</p>
<ol type="1">
<li><p>Samples from 2 different subjects.</p></li>
<li><p>Samples from the same subject, but at 2 different timepoints.</p></li>
<li><p>2 samples taken from the same patient and same day.</p></li>
<li><p>2 sequencing runs of the same sample.</p></li>
</ol>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 x 2
  `Pairwise comparison type`     n
  &lt;chr&gt;                      &lt;int&gt;
1 diff pt                     4463
2 same pt (diff day/sample)   1794
3 same pt/day (diff sample)     34
4 same pt/day/sample            37</code></pre>
</div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<thead>
<tr class="header">
<th style="text-align: left;">Pairwise comparison type</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">diff pt</td>
<td style="text-align: right;">4463</td>
</tr>
<tr class="even">
<td style="text-align: left;">same pt (diff day/sample)</td>
<td style="text-align: right;">1794</td>
</tr>
<tr class="odd">
<td style="text-align: left;">same pt/day (diff sample)</td>
<td style="text-align: right;">34</td>
</tr>
<tr class="even">
<td style="text-align: left;">same pt/day/sample</td>
<td style="text-align: right;">37</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A good distace metric should ideally be capable of demonstrating meaningful microbiome differences between individuals, and changes that occur because of antibiotic disruption. It should not be overly affected by minor differences between samples in a stable individual. We reason that <code>#1</code> and <code>#2</code> comparisons would likely yield differences that vary greatly (i.e.&nbsp;wide range of distances, from 0 to 1). On the other hand, <code>#3</code> and <code>#4</code> comparisons would be expected to be very similar (i.e.&nbsp;low distances nearing 0).</p>
<p>Conventional distance metrics are not ideal when it comes to discerning meaningful differences, as evidenced by the violin plots.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="README_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
<p>We attempted a variety of custom distance functions, where distance was derived by calculating separate distance values across all pairwise samples, after by collapsing by each taxonomic level. The functions shown below were various ways to combined the taxonomic level-specific distances. We find <code>taxhorn.wsk1</code> to be the most optimal combining function, which is the weight mean of distances after excluding Superkingdom level.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="README_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
<p>If we look at hierarchical clustering of samples based on calculated distances, we can see that the weighted average taxhorn puts out a more desirable ordering. For example, note how samples dominated by <em>Enterococcus</em> more naturally clusters together.</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="README_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
<p>asdf</p>
<div class="cell">
<div class="cell-output-display">
<p><img src="README_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>